README.md

```diff
@@ -1,13 +1,408 @@
+Download
+--------
 
-# CPP Collections
+[C++ Collections][dl] (18 KB) -- 10 April 2015
 
-[![Build Status](https://travis-ci.org/natebrennand/cpp_collections.svg?branch=master)](https://travis-ci.org/natebrennand/cpp_collections)
+[dl]: https://github.com/natebrennand/cpp_collections/archive/master.zip
 
-Designing a functional collections library for C++.
 
------------------------
+----
+## Introduction
 
+Collections are fundamental to most programming tasks. More importantly, they let you group and process large sets of data. Whether you  want to create a collection of a financial transactions you want to operate on, or you want to calculate something as summing a list of squares between 0 and 10.
 
-See testing and development guidelines in the `src/` directory.
+![Image of Kesiev](http://i.imgur.com/GnCHqIm.png?1) 
 
+Currently, processing collections in C++ is not only far from perfect, but is syntax-wise far behind many current functional programming paradigms.
 
+Languages are used as tools to communicate, and incidentally, the structure and limitations of a C++ determine the way we think. Learning how to think and within the context of a language provides one the ability to be a proper user, well at the same time 
+
+> In 24 hours you might be able to learn some of the syntax of C++ (if you already know another language), but you couldn't learn much about how to use the language. In short, if you were, say, a Basic programmer, you could learn to write programs in the style of Basic using C++ syntax, but you couldn't learn what C++ is actually good (and bad) for. So what's the point? Alan Perlis once said: "A language that doesn't affect the way you think about programming, is not worth knowing" -- [Peter Norvig][1]
+
+[1]:http://www.norvig.com/21-days.html
+
+This is the principle of linguistic relativity also known as the [Sapir-Whorf hypothesis][2] shows that the constraints of a language are what most often limit our thought processing. This idea, when applied to programming languages shows how Lisp programmers use macros for everything, Python programmers even in Ruby have code that looks like python.  
```

Comment:
Yeah I'll clear it up, I was just trying to convey that languages have limitations in expression.